### PR TITLE
First stab at URL custom type

### PR DESCRIPTION
**Pull request by keithiancolbourne**
Created mutation and successfully tested with this type.

### CODE REVIEW

File: api/src/__tests__/URL.test.js [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+  '///',
+  'http:///a',
+  'foo.com',
+  'rdar://1234',
+  'h://test',
+  'http:// shouldfail.com',
+  'http://foo.bar/foo(bar)baz quux',
+  'ftps://foo.bar/',
+  'http://-error-.invalid/',
+  // 'http://a.b--c.de/',
```
Comment (sastels) [inline api/src/__tests__/URL.test.js]: 🔥

File: api/src/__tests__/URL.test.js [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+  'http://a.b-.co',
+  'http://0.0.0.0',
+  'http://10.1.1.0',
+  'http://10.1.1.255',
+  'http://224.1.1.1',
+  'http://1.1.1.1.1',
+  'http://123.123.123',
+  'http://3628126748',
+  'http://.www.foo.bar/',
+  // 'http://www.foo.bar./',
```
Comment (sastels) [inline api/src/__tests__/URL.test.js]: 🔥

Comment (keithiancolbourne) [reply api/src/__tests__/URL.test.js]: Yeah, I've got to take a closer look at the tests, just wanted to get something started before leaving yesterday. There are a couple not passing.

Comment (sastels) [reply api/src/__tests__/URL.test.js]: Oh, I was referring to deleting the commented out line (or uncomment) :)

Comment (keithiancolbourne) [reply api/src/__tests__/URL.test.js]: ahhhhh...ok! thanks Steve

File: api/src/types/URL.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
+const { GraphQLScalarType } = require('graphql')
+const { GraphQLError } = require('graphql/error')
+const { Kind } = require('graphql/language')
+
+// this file taken and modified from https://github.com/okgrow/graphql-scalars
+
+// see https://gist.github.com/dperini/729294
+const URL_REGEX = new RegExp(
+  /^(?:(?:(?:https?|ftp):)?\/\/)(?:\S+(?::\S*)?@)?(?:(?!(?:10|127)(?:\.\d{1,3}){3})(?!(?:169\.254|192\.168)(?:\.\d{1,3}){2})(?!172\.(?:1[6-9]|2\d|3[0-1])(?:\.\d{1,3}){2})(?:[1-9]\d?|1\d\d|2[01]\d|22[0-3])(?:\.(?:1?\d{1,2}|2[0-4]\d|25[0-5])){2}(?:\.(?:[1-9]\d?|1\d\d|2[0-4]\d|25[0-4]))|(?:(?:[a-z0-9\u00a1-\uffff][a-z0-9\u00a1-\uffff_-]{0,62})?[a-z0-9\u00a1-\uffff]\.)+(?:[a-z\u00a1-\uffff]{2,}\.?))(?::\d{2,5})?(?:[/?#]\S*)?$|^https?:\/\/localhost.*|^http:\/\/(\d{1,3})(\.\d{1,3}){3}(:\d{4})?\/$/i,
```
[the remark this answers is not in the document]
Comment (keithiancolbourne) [reply api/src/types/URL.js]: Can I merge and then change or should I work on changing before merging?

File: api/src/__tests__/URL.test.js [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+  'http://.www.foo.bar/',
+  'http://.www.foo.bar./',
+  'http://10.1.1.1',
+]
+
+describe('URL', () => {
+  describe('valid', () => {
+    // eslint-disable-next-line
+    SHOULD_MATCH.forEach(url => {
+      test(`serialize ${url}`, () => {
```
[the remark this answers is not in the document]
Comment (keithiancolbourne) [reply api/src/__tests__/URL.test.js]: Noted!

File: api/src/types/URL.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (sastels) [reply api/src/types/URL.js]: @sleepycat ☝️ (I'd say let's push and improve later ¯\_(ツ)_/¯ )

**Comment by keithiancolbourne** [comment]
@sleepycat I did not change the regex to match cases like: 

`tcp+unix:///tmp/arangodb.sock
ssl+unix:///tmp/arangodb.sock`

I reckon we can have a chat around how deep we want validation to go.

**Pull request merged**